src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -0,0 +1,875 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
```

Comment:
Some additional test suggestions (if not already covered):

- out dynamic
- out var in interface
- scoping:
```C#
{
    { M(out int x); }
    // cannot use x here
}
```